src/highdicom/sr/templates.py

```diff
@@ -1955,19 +3248,454 @@ def __init__(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Person:
+                attributes = PersonObserverIdentifyingAttributes.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            else:
+                raise ValueError('Unexpected observer type "{item.meaning}".')
+            context = ObserverContext(
+                observer_type=item.value,
+                observer_identifying_attributes=attributes
+            )
+            observer_contexts.append(context)
+        return observer_contexts
+
+    def get_subject_contexts(
+        self,
+        subject_class: Optional[CodedConcept] = None
+    ) -> List[SubjectContext]:
+        """Get subject contexts.
+
+        Parameters
+        ----------
+        subject_class: Union[highdicom.sr.coding.CodedConcept, pydicom.sr.coding.Code], optional
+            Type of subject ("Specimen", "Fetus", or "Device") for which should
+            be filtered
+
+        Returns
+        -------
+        List[highdicom.sr.SubjectContext]
+           Subject contexts
+
+        """  # noqa: E501
+        root_item = self[0]
+        matches = [
+            (i + 1, item) for i, item in enumerate(root_item.ContentSequence)
+            if item.name == codes.DCM.SubjectClass
+        ]
+        subject_contexts = []
+        for i, (index, item) in enumerate(matches):
+            if subject_class is not None:
+                if item.value != subject_class:
+                    continue
+            try:
+                next_index = matches[i + 1][0]
+            except IndexError:
+                next_index = -1
+            if item.value == codes.DCM.Specimen:
+                attributes = SubjectContextSpecimen.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Fetus:
+                attributes = SubjectContextFetus.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Device:
+                attributes = SubjectContextDevice(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            else:
+                raise ValueError('Unexpected subject class "{item.meaning}".')
+            context = SubjectContext(
+                subject_class=item.value,
+                subject_class_specific_context=attributes
+            )
+            subject_contexts.append(context)
+        return subject_contexts
+
+    def get_planar_roi_measurement_groups(
+        self,
+        tracking_uid: Optional[str] = None,
+        finding_type: Optional[Union[CodedConcept, Code]] = None,
+        finding_site: Optional[Union[CodedConcept, Code]] = None,
```

Comment:
I think filtering by finding site may require bit more thought. In the "encoding" API (and the "finding_sites" property of the decoding API), finding sites are specified using their own `FindingSite` class, which allows for laterality and topographical modifiers. But here we are just checking the name against a coded concept. Should this method accept a `FindingSite` object, and include the modifier and laterality in the definition of a match, so that you can search for the "left lung" for example? Perhaps we should add a `__eq__` method on `FindingSite`?

Additionally I tried search some of my SRs by tracking UID and this method just returned an empty list even when the finding site matched.

Reply:
I think the problem with the finding site match has been fixed via b3316a47b3a185133e331e6593b97447e2bd8725.

I like the idea of passing a `FindingSite` object and thereby also considering the laterality modifier. Adding `__eq__` would be a convenient way to achieve that. Now that I think about it, we should probably override `__eq__` on most if not all objects in the `content` module(s).

Reply:
The problem with the matching has been resolved.